src/main/java/plugily/projects/villagedefense/user/data/FileStats.java

```diff
@@ -117,8 +117,13 @@ private void updateStats(User user) {
     String uuid = user.getUniqueId().toString();
 
     for(StatsStorage.StatisticType stat : StatsStorage.StatisticType.values()) {
-      if(stat.isPersistent()) {
-        config.set(uuid + "." + stat.getName(), user.getStat(stat));
+      if(!stat.isPersistent()) {
+        continue;
+      }
+      String path = user.getUniqueId().toString() + "." + stat.getName();
+      int value = user.getStat(stat);
+      if (value > 0 || config.contains(path)) {
```

Comment:
I don't think the `contains` 2nd condition is required.

Reply:
It's future-proof. I don't know if you guys will add some stats with negative values.

Reply:
Or it's required when you have `SetLevelArgument`. They may set the level of their players back to 0.